chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -0,0 +1,355 @@
+	 *     dataProvider: (item, dataKey) => {
+	 *         // item: the current item in the row
+	 *         // dataKey: the dataKey of the column
+	 *         // return: the data to display in the column
+	 *         return item.getField('title').split('').reverse().join('');
+	 *     },
+	 *     zoteroPersist: ['width', 'hidden', 'sortDirection'], // persist the column properties
+	 * });
+	 * ```
+	 * @example
+	 * Register multiple custom columns:
+	 * ```js
+	 * const registeredDataKeys = await Zotero.ItemTreeManager.registerColumns(
+	 * [
+	 *     {
+	 *          dataKey: 'rtitle',
+	 *          iconPath: 'chrome://zotero/skin/tick.png',
+	 *          label: 'reversed title',
+	 *          pluginID: 'make-it-red@zotero.org', // Replace with your plugin ID
+	 *          dataProvider: (item, dataKey) => {
+	 *              return item.getField('title').split('').reverse().join('');
+	 *          },
+	 *     },
+	 *     {
+	 *          dataKey: 'utitle',
+	 *          label: 'uppercase title',
+	 *          pluginID: 'make-it-red@zotero.org', // Replace with your plugin ID
+	 *          dataProvider: (item, dataKey) => {
+	 *              return item.getField('title').toUpperCase();
+	 *          },
+	 *     },
+	 * ]);
+	 * ```
+	 */
+	async registerColumns(options) {
+		const registeredDataKeys = this._addColumns(options);
+		if (!registeredDataKeys) {
+			return false;
+		}
+		this._addPluginShutdownObserver();
+		await this._notifyItemTrees();
+		return registeredDataKeys;
+	}
+
+	/**
+	 * Unregister a custom column.
+	 * Although it's async, resolving does not promise the item trees are updated.
+	 * @param {string | string[]} dataKeys - The dataKey of the column to unregister
+	 * @returns {boolean} true if the column(s) are unregistered
+	 * @example
+	 * ```js
+	 * Zotero.ItemTreeManager.unregisterColumns('rtitle');
+	 * ```
+	 */
+	async unregisterColumns(dataKeys) {
+		const success = this._removeColumns(dataKeys);
+		if (!success) {
+			return false;
+		}
+		await this._notifyItemTrees();
+		return true;
+	}
+
+	// TODO: add cell renderer registration
+
+	/**
+	 * Get column(s) that matches the properties of option
+	 * @param {string | string[]} [enabledTreeIDs] - The tree IDs to match
+	 * @param {ItemTreeCustomColumnFilters} [options] - An option or array of options to match
+	 * @returns {ItemTreeCustomColumnOptions[]}
+	 */
+	getCustomColumns(enabledTreeIDs, options) {
+		const allColumns = Object.values(this._customColumns).map(opt => Object.assign({}, opt));
+		if (!enabledTreeIDs && !options) {
+			return allColumns;
+		}
+		let filteredColumns = allColumns;
+		if (enabledTreeIDs) {
+			if (typeof enabledTreeIDs === "string") {
+				enabledTreeIDs = [enabledTreeIDs];
+			}
+			filteredColumns = filteredColumns.filter(col => enabledTreeIDs.includes("*") || enabledTreeIDs.find(treeID => (col.enabledTreeIDs || ["main"]).includes(treeID)));
```

Comment:
Unroll this to be more readable. e.g. separate conditional for *, set default col.enabledTreeIDs = ["main"] upon registration.